src/Connector.js

```diff
@@ -279,6 +275,17 @@ class Connector {
     return client;
   }
 
+  async connectDevices() {
+    const [topic, payload] = await this.setupDevices();
+    await this.messageHandler(topic, payload);
+  }
+
+  // FIXME: doesnt trigger client.on('message', (topic, payload))
+  async setupDevices() {
+    const devices = await this.listDevices();
+    return promisify(this.client, 'message', subscribeToEntities, this.client, devices);
```

Comment:
The promisify function is wrapping a method `once` and not a `on` method.
In other words when you receive the first message it will stop to listen the others

Reply:
Maybe this could be related from what you've comment here: https://github.com/CESARBR/knot-fog-connector-fiware/pull/12#discussion_r351329510

Reply:
Note though, that the method `subscribeToEntities` is async and inside the function `promisify` you are not calling the `method` with `await`

Reply:
@ramonhpr, kindly take a look at [this](https://github.com/CESARBR/knot-fog-connector-fiware/pull/12#discussion_r352818172) comment :innocent: